lib/ohai/plugins/aix/virtualization.rb

```diff
@@ -32,17 +32,19 @@
       virtualization[:lpar_name] = lpar_name
     end
 
-    so = shell_out("uname -W")
-    wpar_no = so.stdout.split($/)[0]
+    wpar_no = shell_out("uname -W").stdout.chomp
     if wpar_no.to_i > 0
       virtualization[:wpar_no] = wpar_no
     else
       # the below parses the output of lswpar in the long format
       so = shell_out("lswpar -L").stdout.scan(/={65}.*?(?:EXPORTED\n\n)+/m)
       wpars = Mash.new
       so.each do |wpar|
-        wpar_name = wpar.lines[1].split[0]
+        wpar_heading = wpar.lines[1].split
```

Comment:
You could potentially simplify this a bit by using `.split(" - ")`.

Reply:
I'm a bit nervous about relying on particular space counts. I had it that way originally and then went back to this one.